### PR TITLE
Un stake improvements

### PR DESCRIPTION
**Pull request by sasurobert**
- removed unnecessary fields in delegation struct
- replaced unbonding period from nonce to epoch as to allow easier computation

### CODE REVIEW

File: cmd/node/config/systemSmartContractsConfig.toml [hunk 1]
```diff
@@ -3,6 +3,7 @@
     MinStakeValue = "100000000000000000000" #100 eGLD
     MinUnstakeTokensValue = "10000000000000000000" #10eGLD
     UnBondPeriod = 250
+    UnBondPeriodInEpochs = 10
```
Comment (iulianpascalau) [inline cmd/node/config/systemSmartContractsConfig.toml:6]: can we set the default to 1 here?

File: vm/systemSmartContracts/delegation.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ const serviceFeeKey = "serviceFee"
 const totalActiveKey = "totalActive"
 const rewardKeyPrefix = "reward"
 const fundKeyPrefix = "fund"
+const maxUnStakedFunds = 50
```
Comment (raduchis) [inline vm/systemSmartContracts/delegation.go:29]: this is an actual len, not the funds.

**Review by raduchis — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by raduchis — APPROVED** [review]

File: cmd/node/config/systemSmartContractsConfig.toml [hunk 3]
```diff
@@ -3,6 +3,7 @@
     MinStakeValue = "100000000000000000000" #100 eGLD
     MinUnstakeTokensValue = "10000000000000000000" #10eGLD
     UnBondPeriod = 250
+    UnBondPeriodInEpochs = 1
```
Comment (SebastianMarian) [inline cmd/node/config/systemSmartContractsConfig.toml:6]: This should be set to 10 for mainnet

File: vm/systemSmartContracts/delegation.go (hunk 2 above)
Comment (SebastianMarian) [reply vm/systemSmartContracts/delegation.go:29]: maxNumOfUnStakes or maxNumOfUnStakedFunds ?

File: cmd/node/config/systemSmartContractsConfig.toml (hunk 3 above)
Comment (iulianpascalau) [reply cmd/node/config/systemSmartContractsConfig.toml:6]: Of course

File: vm/systemSmartContracts/delegation.go (hunk 2 above)
Comment (sasurobert) [reply vm/systemSmartContracts/delegation.go]: pushed

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]

**Pull request merged**